plutus-pab/src/Plutus/PAB/App.hs

```diff
@@ -139,24 +141,30 @@ appEffectHandlers config trace =
 
 runApp ::
     forall a.
-    Trace IO (PABLogMsg ContractExe) -- ^ Top-level tracer
+    EventfulBackend
+    -> Trace IO (PABLogMsg ContractExe) -- ^ Top-level tracer
     -> Config -- ^ Client configuration
     -> App a -- ^ Action
     -> IO (Either PABError a)
-runApp trace config@Config{endpointTimeout} = Core.runPAB (Timeout endpointTimeout) (appEffectHandlers config trace)
+runApp eventfulBackend trace config@Config{endpointTimeout} = Core.runPAB (Timeout endpointTimeout) (appEffectHandlers eventfulBackend config trace)
 
 type App a = PABAction ContractExe AppEnv a
 
-mkEnv :: Trace IO (PABLogMsg ContractExe) -> Config -> IO AppEnv
-mkEnv appTrace appConfig@Config { dbConfig
+data EventfulBackend = SqliteBackend | InMemoryBackend
+    deriving (Eq, Ord, Show)
+
+mkEnv :: EventfulBackend -> Trace IO (PABLogMsg ContractExe) -> Config -> IO AppEnv
+mkEnv eventfulBackend appTrace appConfig@Config { dbConfig
              , nodeServerConfig =  MockServerConfig{mscBaseUrl, mscSocketPath, mscSlotConfig}
              , walletServerConfig
              , chainIndexConfig
              } = do
     walletClientEnv <- clientEnv (Wallet.baseUrl walletServerConfig)
     nodeClientEnv <- clientEnv mscBaseUrl
     chainIndexEnv <- clientEnv (ChainIndex.ciBaseUrl chainIndexConfig)
-    dbConnection <-  dbConnect appTrace dbConfig
+    dbConnection <- case eventfulBackend of
+        SqliteBackend   -> Left <$> dbConnect appTrace dbConfig
```

Comment:
Now that explains my previous question about the `fmap Left` - I would really prefer the use of a custom sumtype here.

I know Left/Right can have any meaning but practically speaking `Left` is "something went wrong" 99% of the time. This made the code above so weird to me.

Could we just have 
```haskell
data EventfulBackendMode = EventfulSqlite | EventfulInMemory
```
or something similar?